newrelic/open-install-library
# feat(cassandra): Env vars

Updates the Cassandra recipes to be able to accept required user input as environment variables. This is necessary so that we can add Cassandra as a target to `ansible-install`.

**Notes**
* Tested on debian10 and AL2 during development

**Testing**
_Happy path_
1. Log in to machine that is running Cassandra
2. Clone this repo and switch to this PR branch
3. Run the installation with `--localRecipes /path/to/recipes`
4. Use the standard test credentials when prompted

_Hostname_
1. You can test the hostname input behavior by providing `0.0.0.0` as an invalid value and running with `--localRecipes`

_Custom port_
1. Search for `JMX_PORT` in `cassandra-env.sh` and change it from `7199`. Find the script on your host at:
   ```
   // Debian
   /etc/cassandra

   // RHEL
   /etc/cassandra/conf
   ```
2. Restart Cassandra
   ```
   sudo service casssandra restart
   ```
3. Run the installation with `--localRecipes` and the CLI should prompt you for a port

_Install in non-interactive mode with environment variables_
1. Add the environment variables to your installation command as well as the `-y` flag
   ```
   ... sudo NEW_RELIC_CASSANDRA_USERNAME=<username> NEW_RELIC_CASSANDRA_PASSWORD=<password> NEW_RELIC_CASSANDRA_PORT=<port> NEW_RELIC_CASSANDRA_HOSTNAME=<hostname> ... /usr/local/bin newrelic install -n cassandra-open-source-integration --localRecipes /path/to/recipes -y
   ```
2. To test invalid inputs, add the `--debug` flag to see error output

_Legacy environment variables_
```
NR_CLI_DB_USERNAME
NR_CLI_DB_PASSWORD
NR_CLI_DB_HOSTNAME
```
1. You should be able to use the new and legacy environment variables interchangeably.
